modules/apps/app-builder/app-builder-test/src/testFunctional/tests/AppBuilder.testcase

```diff
@@ -308,4 +308,191 @@ definition {
 		AppBuilderAdmin.assertNewButtonIsVisible(buttonValue = "New Table View");
 	}
 
+	@description = "Validate returns by search engine using a search term"
```

Comment:
Hey Tai, I would suggest you to create a macro "searchByItem" at AppBuilderAdmin.macro file where you can move these two actions Type and Click to this macro. 

Look that you are using this piece of code in every test case that performs a search, so you will be increasing the code reuse creating a this macro instead of copy and paste this piece in every place you want to do this action.

I would suggest you to add the xpaths that you are using in the AppBuilder.path so you will be encapuslating the xpaths in the right place.

Additionally, I would suggest you to change your xpaths to:
at Type action: locator1 = "//input[contains(@class,'input-group')]"
at Click action: locator1 = "//*[contains(@class,'lexicon-icon-search')]"

For the first one (Type) in general, the class attributes are less frequent changed, so your xpath will be more robust if you use this kind of element instead of texts.
For the second one (Click), it is good to = (equals operator) when we have a single element or id, but when we have some elements that contains blank spaces between the names, if a dev change one of this elements or remove the space, our xpath will no longer work. So replacing it to contains and adding a part of the class name you are looking for could be a solution.

So, your xpath at AppBuilder.path file will be something like this:

<tr>
	<td>SEARCH_BUTTON</td>
	<td>//*[contains(@class,'lexicon-icon-search')]</td>
	<td></td>
</tr>

<tr>
	<td>SEARCH_BAR_INPUT</td>
	<td>//input[contains(@class,'input-group')]</td>
	<td></td>
</tr>

and you will call these paths into your macro like this:

macro searchByItem {
	Type(
			locator1 = "AppBuilder#SEARCH_BAR_INPUT",
			value1 = "${item}");
	
	Click(locator1 = "AppBuilder#SEARCH_BUTTON");
}

and you will use it in the AppBuilder.testcase like this:

searchByItem(item = "Apple");


and I would suggest you do the same for the validation of the search results (for positive (item is found) and negative scenarios (item is not found)) so you need to create a path and a macro to replace the following pieces of code:

		AssertTextEquals(
			locator1 = "//div[contains(.,'Results')]//li[@class='tbar-item tbar-item-expand']",
			value1 = "0 Results for Apple");

		AssertTextEquals(
			locator1 = "//h1[contains(@class,'taglib-empty-result-message-title')]",
			value1 = "No results were found.");

I would also suggest to create a path and a macro to this one:
Click(locator1 = "//button[contains(text(),'Clear')]");

and for this one:
AssertTextEquals(
			locator1 = "//tr[@data-testid='item'][1]//a",
			value1 = "Date Field - Form View Title");

where you can pass the [1], [2] or [N] by parameter to be inserted in the locator as well as the value that you are looking for.